lib/content-services/src/lib/search/components/search-chip-autocomplete-input/search-chip-autocomplete-input.component.ts

```diff
@@ -0,0 +1,129 @@
+/*!
+ * @license
+ * Copyright © 2005-2023 Hyland Software, Inc. and its affiliates. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Component, ViewEncapsulation, ElementRef, ViewChild, OnInit, OnDestroy, Input, Output, EventEmitter, Pipe, PipeTransform } from '@angular/core';
+import { ENTER } from '@angular/cdk/keycodes';
+import { FormControl } from '@angular/forms';
+import { MatAutocompleteSelectedEvent } from '@angular/material/autocomplete';
+import { MatChipInputEvent } from '@angular/material/chips';
+import { Observable, Subject } from 'rxjs';
+import { map, startWith, takeUntil } from 'rxjs/operators';
+
+@Pipe({
+    name: 'isIncluded'
+})
+export class IsIncludedPipe implements PipeTransform {
+    transform(value: string, array: string[]): boolean {
+        return array.includes(value);
+    }
+}
+
+@Component({
+    selector: 'adf-search-chip-autocomplete-input',
+    templateUrl: './search-chip-autocomplete-input.component.html',
+    styleUrls: ['./search-chip-autocomplete-input.component.scss'],
+    encapsulation: ViewEncapsulation.None
+})
+export class SearchChipAutocompleteInputComponent implements OnInit, OnDestroy {
+    @ViewChild('optionInput')
+    optionInput: ElementRef<HTMLInputElement>;
+
+    @Input()
+    autocompleteOptions: string[] = [];
+
+    @Input()
+    onReset$: Observable<void>;
+
+    @Input()
+    allowOnlyPredefinedValues = true;
+
+    @Output()
+    optionsChanged: EventEmitter<string[]> = new EventEmitter();
+
+    readonly separatorKeysCodes = [ENTER] as const;
+    formCtrl = new FormControl('');
+    filteredOptions$: Observable<string[]>;
+    selectedOptions: string[] = [];
+    private onDestroy$ = new Subject<void>();
+
+    constructor() {
+        this.filteredOptions$ = this.formCtrl.valueChanges.pipe(
+            startWith(null),
```

Comment:
startWith seems deprecated: 
<img width="814" alt="image" src="https://github.com/Alfresco/alfresco-ng2-components/assets/115619721/4e0a2fbb-6adf-4db3-9336-7ab3364e32ad">

Reply:
It looks like it is actually not deprecated, take a look https://stackoverflow.com/questions/56571406/is-the-startwith-operator-in-rxjs-really-deprecated

Reply:
Hmm wondering if these answers from stackoverflow are still valid. For example some of them are from 2021 year or older. But it is small comment so we can leave it for now